### ISSUE TITLE
Adding invite to IRCClient

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4820]
|<img alt="taos's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| taos reported|
|-|-|
|Trac ID|trac#4820|
|Type|enhancement|
|Created|2011-01-24 14:26:27Z|
|Branch|https://github.com/twisted/twisted/tree/ircclient-invite-4820|

I think there should be a invite method to IRCClient. 

Attachments:

* [invite.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/21a/21a62ff2e0360021799669cde2acfed97073029a/f6c1b9dc6a3a40ea6c282efc2ff76aa372cb9c03.patch) (2297 bytes) - added by taos on 2011-01-24 14:26:40Z - 
* [4820.feature](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/21a/21a62ff2e0360021799669cde2acfed97073029a/f06366e3581499f636de7ce95122c827b43a5ef5.feature) (64 bytes) - added by taos on 2011-01-24 14:29:48Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4820 4820
type__enhancement enhancement
reporter__taos taos
priority__normal normal
milestone__ 
branch__branches_ircclient_invite_4820 branches/ircclient-invite-4820
branch_author__taos taos
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1295879187000000 1295879187000000
changetime__1296266026000000 1296266026000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="taos's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|taos removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [30577]) Branching to 'ircclient-invite-4820'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [30578]) Apply invite.patch

refs [#4820](https://github.com/twisted/twisted/issues/4820)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

Thanks for working on this patch.  The implementation looks straightforward and correct to me.  Concerning the tests though:

  1. I like `ClientInviteTests`, except for
    1. the lack of docstrings
    1. the `testFoo` naming (instead of the preferred `test_foo` style).
  1. Oops, also, the `setUp` call in `testInvite` isn't necessary.  I think `ClientTests.test_invite` is probably redundant with the tests on `ClientInviteTests` though.
  1. methods should be separated from each other by two blank lines; classes by three.
  1. New APIs (like `IRCClient.invite`) should be marked up with `@since` tags

These are simple points, so I'll go ahead and correct them and then merge the change.  Thanks again!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

(In [30580]) vertical whitespace fixes

refs [#4820](https://github.com/twisted/twisted/issues/4820)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

(In [30581]) @since marker

refs [#4820](https://github.com/twisted/twisted/issues/4820)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun commented|
|-|-|

(In [30582]) Add docstrings to everything on ClientInviteTests; fix vertical whitespace and test method naming; remove spurious setUp call; insert missing whitespace between function arguments.

refs [#4820](https://github.com/twisted/twisted/issues/4820)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun commented|
|-|-|

(In [30583]) Remove the redundant test_invite

refs [#4820](https://github.com/twisted/twisted/issues/4820)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun commented|
|-|-|

(In [30584]) restore deleted blank line

refs [#4820](https://github.com/twisted/twisted/issues/4820)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_11"></a>|@exarkun set status to `closed`|
|-|-|

(In [30585]) Merge ircclient-invite-4820

Author: taos, exarkun
Reviewer: exarkun
Fixes: [#4820](https://github.com/twisted/twisted/issues/4820)

Add `IRCClient.invite` method to send INVITE messages.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
